File: src/main/java/me/alegian/thavma/impl/common/block/entity/BEHelper.java
package me.alegian.thavma.impl.common.block.entity;

import net.minecraft.core.BlockPos;
import net.minecraft.world.level.Level;
import net.minecraft.world.level.block.Block;

public class BEHelper {
  public static void updateServerBlockEntity(Level level, BlockPos blockPos) {
    var be = level.getBlockEntity(blockPos);
    if (be != null) {
      be.setChanged();
      level.sendBlockUpdated(blockPos, be.getBlockState(), be.getBlockState(), Block.UPDATE_CLIENTS);
    }
  }
}

File: src/main/java/me/alegian/thavma/impl/common/block/entity/BEHelper.kt
package me.alegian.thavma.impl.common.block.entity

import net.minecraft.core.BlockPos
import net.minecraft.world.level.Level
import net.minecraft.world.level.block.Block

fun Level.updateServerBlockEntity(blockPos: BlockPos) {
    getBlockEntity(blockPos)?.let { be ->
        be.setChanged()
        sendBlockUpdated(blockPos, be.blockState, be.blockState, Block.UPDATE_CLIENTS)
    }
}

